src/main/java/dev/morling/onebrc/CalculateAverage_jhonesto.java

```diff
@@ -0,0 +1,396 @@
+                sum = max = min = t;
+                firstMeasure = false;
+                count = ONE;
+                return;
+            }
+
+            max = max < t ? t : max;
+            min = min > t ? t : min;
+            sum += t;
+            count += ONE;
+
+        }
+
+        @Override
+        public String toString() {
+
+            return String.format("%s=%.1f/%.1f/%.1f",
+                    new String(station, StandardCharsets.UTF_8).trim(),
+                    Math.round(min) / TEN_DOUBLE,
+                    Math.round(sum / (double) count) / TEN_DOUBLE,
+                    Math.round(max) / TEN_DOUBLE);
+        }
+
+        public int compareTo(Measurement other) {
+            return new String(this.station).compareTo(new String(other.station));
+        }
+
+    }
+
+    // +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+    // ++++++++++++++++++++++++++++ UTILS ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+    // +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+    // Get the end of partitions; b -> buffer, mbs -> max_buffer_size
+    private static long[] getEOP(int[] b, int mbs, File file) {
+
+        long eof[] = new long[b.length];
+
+        try (RandomAccessFile raf = new RandomAccessFile(file, READ)) {
+
+            for (int i = ZERO; i < b.length; i++) {
+
+                long size = ((long) mbs * i) +
+                        (long) (i == b.length - ONE ? b[i] : mbs);
+
+                raf.seek((size - ONE));
+
+                while (raf.read() != LINE_FEED) {
+                    raf.seek(--size - ONE);
+                }
+
+                eof[i] = size;
+            }
+        }
+        catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        return eof;
+    }
+
+    public static byte[] asByteArrayCopy(byte[] arr, int length) {
+        byte[] b = new byte[length];
+        System.arraycopy(arr, ZERO, b, ZERO, length);
+        return b;
+    }
+
+    private static void resolveMeasurement(byte[] measurement, HashMap<Long, Measurement> table) {
+
+        short temperature = 0;
+
+        int j = measurement.length - ONE;
+
+        for (short exp = ZERO; j > ZERO; j -= ONE, exp += ONE) {
+
+            if (measurement[j] == SEMICOLON) {
+                break;
+            }
+
+            if (measurement[j] == MINUS) {
+                temperature *= -ONE;
+            }
+            else if (measurement[j] == PERIOD) {
+                exp -= ONE;
+            }
+            else {
+                temperature += (short) ((measurement[j] % MOD_NUMBER) * pow(RADIX, exp));
+            }
+
+        }
+
+        byte[] station = asByteArrayCopy(measurement, j);
+
+        long hash = getHash(station);
+
+        Measurement m = table.get(hash);
+
+        if (m != null) {
```

Comment:
Thank you so much Roy! 

Now it's clear to me. I was thinking in a completely opposite direction.

Should I close this pull request?

And What timezone are you using? Because, I am in UTC-3 and have other things to do here. 
I'm not sure if I can deliver on time.

I really appreciate your help.